worlds/stardew_valley/stardew_rule/state.py

```diff
@@ -47,7 +47,7 @@ def __repr__(self):
         return f"Received {self.count} {self.items}"
 
 
-@dataclass(frozen=True)
+@dataclass(frozen=True, slots=True)
```

Comment:
What does this do?

Reply:
That does break python 3.8, and it's removed now.

More seriously, it changes how the python object is created to use a C style struct for the fields instead of the usual python dictionary. It speeds up the creation of the objects, at the cost of some flexibility.